File: codegen/results/v002/qwen3.5_4b_nothink/20260412_070316/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
)

// calculateDiscount calculates a bulk discount
func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	// Helper function to run a test case
	runTest := func(price, quantity int, expected float64, description string) {
		result := calculateDiscount(float64(price), quantity)
		if result == expected {
			fmt.Printf("PASS: %s\n", description)
		} else {
			fmt.Printf("FAIL: %s (expected: %.6f, got: %.6f)\n", description, expected, result)
		}
	}

	// Test case 1: quantity < 10
	runTest(10.0, 5, 50.0, "quantity 5 with no discount")

	// Test case 2: boundary case quantity = 9
	runTest(10.0, 9, 90.0, "quantity 9 (boundary, no discount)")

	// Test case 3: boundary case quantity = 10
	runTest(10.0, 10, 90.0, "quantity 10 (boundary, 10% discount)")

	// Test case 4: quantity just below 50
	runTest(10.0, 49, 441.0, "quantity 49 (boundary, 10% discount)")

	// Test case 5: boundary case quantity = 50
	runTest(10.0, 50, 375.0, "quantity 50 (boundary, 25% discount)")
}

File: codegen/results/v002/qwen3.5_4b_nothink/20260412_070316/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Test cases
const testCases = [
    {
        description: "Quantity less than 10 (boundary: 9)",
        price: 10,
        quantity: 9,
        expected: 90
    },
    {
        description: "Quantity between 10 and 49 (boundary: 10)",
        price: 10,
        quantity: 10,
        expected: 90
    },
    {
        description: "Quantity near boundary 50 (boundary: 49)",
        price: 10,
        quantity: 49,
        expected: 441
    },
    {
        description: "Quantity at boundary 50",
        price: 10,
        quantity: 50,
        expected: 375
    }
];

// Run tests
testCases.forEach((tc) => {
    const result = calculateDiscount(tc.price, tc.quantity);
    const passed = result === tc.expected;
    
    if (passed) {
        console.log(`PASS: ${tc.description}`);
    } else {
        console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${result})`);
    }
});